llama-index-core/llama_index/core/callbacks/base.py

```diff
@@ -68,6 +68,15 @@ def __init__(self, handlers: Optional[List[BaseCallbackHandler]] = None):
                     )
             handlers.append(new_handler)
 
+        # if we passed in no handlers, use the global default
+        if len(handlers) == 0:
+            from llama_index.core.settings import Settings
+
+            # hidden var access to prevent recursion in getter

```

Comment:
oh interesting..

Reply:
Yea just due to how the setter works (its lazily initialized)